app/src/ui/lib/list/list.tsx

```diff
@@ -1188,7 +1188,7 @@ export class List extends React.Component<IListProps, IListState> {
         <ListRow
           key={params.key}
           id={id}
-          role={this.props.role === undefined ? undefined : 'listitem'}
```

Comment:
In ListRow the logic is:

```
sectionHasHeader && rowIndex.row === 0
            ? 'presentation'
            : role ?? 'option'
```

List logic:
``` role={this.props.role ?? 'listbox'} ```

So our two use cases are:
1. CommitList which now does define a role as either `list` or `listbox`.  (History or unreachable commit list)
2. Every other list where role = undefined.

For every other, we want  to keep current behavior as `listbox` and `option`. This does uphold, if role is undefined, the main grid is "defined or listbox", this logic here says undefined !== list so the ListBox will receive undefined. which will then filter through the sectionHeader logic to the same "definedRole or option".

For the CommitList, it will send in a defined role of `list` or listbox` so it will pass that and set it on the main List. and this condition will say if it `list` to define a role for the list item as `listitem` otherwise do undefined which will again follow the above and use the default of `option` assuming it isn't a sectionHeader row.